python-range/float_range.py

```diff
@@ -0,0 +1,97 @@
+import math
+from dataclasses import dataclass, field
+
+
+@dataclass
+class FloatRange:
+    start: float | int
+    stop: float | int = None
+    step: float | int = 1.0
+
+    def __post_init__(self):
+        """Validate parameters."""
+        # Only one argument is given
+        if self.stop is None:
+            self.stop = self.start
+            self.start = 0
+
```

Comment:
If step is zero, we should raise an error to handle edge cases:

```python
if math.isclose(self.step, 0):
    raise ValueError("'step' must not be zero")
```

Reply:
This is checked by `if not isinstance(self.step, float | int) or self.step == 0` below. I'm switching to `isclose()`.